lib/rails_core_extensions/translations.rb

```diff
@@ -0,0 +1,44 @@
+module RailsCoreExtensions
+  module Translations
+    def self.included(base)
+      base.extend ClassMethods
+    end
+
+    module ClassMethods
+      def translate(key, options = {})
+        I18n.translate key, options.merge(scope: translation_key)
+      end
+
+      def translation_key
+        (base_activerecord_class || base_ruby_class).name.tableize.singularize
+      end
+
+      def base_activerecord_class
+        return unless defined?(ActiveRecord) && is_a?(ActiveRecord::Base)
```

Comment:
Just a small suggestion, I personally like to explicitly declare the ``nil`` return value for functional methods
```ruby
return nil unless defined?(ActiveRecord) && is_a?(ActiveRecord::Base)
```